pkg/controller/ttlafterfinished/ttlafterfinished_controller.go

```diff
@@ -243,14 +252,15 @@ func (tc *Controller) processJob(ctx context.Context, key string) error {
 
 // processTTL checks whether a given Job's TTL has expired, and add it to the queue after the TTL is expected to expire
 // if the TTL will expire later.
-func (tc *Controller) processTTL(job *batch.Job) (expiredAt *time.Time, err error) {
+func (tc *Controller) processTTL(ctx context.Context, job *batch.Job) (expiredAt *time.Time, err error) {
```

Comment:
Looking at this function, it seems more suitable to pass a logger here. It's a local helper which does do any communication which might need a context.